### DISCUSSION TITLE
Is it possible to set an "infinite" timeout?

### DISCUSSION THREAD

**Discussion by KaranAhlawat** [joaotavora/eglot#1307] [Q&A] [answered]
Related to what I was doing in  #1305, the use case is that Metals sends a `window/showMessageRequest' request, and eglot handles that through completing-read (beautiful btw).

However, if the user doesn't pick anything within 10 seconds, the RPC request is timed out, and then the user needs to start over again.

I'd like to be essentially disable the `timeout` in certain cases, where the user needs to provide input or select from a list of choices. I tried setting the `:timeout` key to `nil` in `eglot--request`, but that clearly wasn't the way.

If this isn't possible, would setting the timeout to a high value (10 minutes? 1 hour?) be a good alternative?

**Comment by joaotavora** [comment]
This is related to the fact that there is a server request within a client request.  I think it should be considered a bug in Eglot.

I don't know exactly how to best solve it though.  Somehow maybe the timeout timer should be stopped while `completing-read` is waiting for the user to answer.  Don't know if that's possible or easy to do, though.

**Reply by joaotavora** [reply, marked as answer]
> to nil in eglot--request, but that clearly wasn't the way.

That doesn't disable the timeout I think, just makes it use the default value.  Try `:timeout some-big-number`

**Reply by KaranAhlawat** [reply]
Alright, I'll take that as an answer and keep on going. Thank you!
